src/codemagic/tools/app_store_connect/action_groups/builds_action_group.py

```diff
@@ -71,6 +75,45 @@ def get_build(self, build_id: ResourceId, should_print: bool = True) -> Build:
 
         return self._get_resource(build_id, self.api_client.builds, should_print)
 
+    @cli.action(
+        "list",
+        AppArgument.APPLICATION_ID_RESOURCE_ID_OPTIONAL,
+        *ArgumentGroups.LIST_BUILDS_FILTERING_ARGUMENTS,
+        action_group=AppStoreConnectActionGroup.BUILDS,
+        deprecation_info=cli.ActionDeprecationInfo("list-builds", "0.49.0"),
+    )
+    def list_builds(
+        self,
+        application_id: Optional[ResourceId] = None,
+        expired: Optional[bool] = None,
+        not_expired: Optional[bool] = None,
+        build_id: Optional[ResourceId] = None,
+        pre_release_version: Optional[str] = None,
+        processing_state: Optional[BuildProcessingState] = None,
+        beta_review_state: Optional[Union[BetaReviewState, Sequence[BetaReviewState]]] = None,
+        build_version_number: Optional[int] = None,
+        should_print: bool = True,
+    ) -> List[Build]:
+        """
+        List Builds from Apple Developer Portal matching given constraints
+        """
+        try:
+            expired_value = Argument.resolve_optional_two_way_switch(expired, not_expired)
+        except ValueError:
+            flags = f"{BuildArgument.EXPIRED.flag!r} and {BuildArgument.NOT_EXPIRED.flag!r}"
+            raise BuildArgument.NOT_EXPIRED.raise_argument_error(f"Using mutually exclusive switches {flags}.")
+
+        builds_filter = self.api_client.builds.Filter(
+            app=application_id,
+            expired=expired_value,
+            id=build_id,
+            processing_state=processing_state,
+            beta_app_review_submission_beta_review_state=beta_review_state,
+            version=build_version_number,
+            pre_release_version_version=pre_release_version,
+        )
+        return self._list_resources(builds_filter, self.api_client.builds, should_print)
```

Comment:
Used to be defined in https://github.com/codemagic-ci-cd/cli-tools/blob/3670fa624997f17cb437bc88198f7bdee3594f35/src/codemagic/tools/app_store_connect.py#L267-L297